api/src/Modules/registration/Controller/Ticket/BaseTicket.php

```diff
@@ -0,0 +1,155 @@
+<?php declare(strict_types=1);
+/*.
+    require_module 'standard';
+.*/
+
+namespace App\Modules\registration\Controller\Ticket;
+
+use Slim\Container;
+use Slim\Http\Request;
+use Slim\Http\Response;
+use App\Modules\registration\Controller\BaseRegistration;
+
+abstract class BaseTicket extends BaseRegistration
+{
+
+    public $id = 0;
+
+
+    public function __construct(Container $container)
+    {
+        parent::__construct($container);
+
+    }
+
+
+    protected function buildTicketHateoas(Request $request, $id)
+    {
+        if ($this->id !== 0) {
+            $path = $request->getUri()->getBaseUrl();
+            $this->addHateoasLink('self', $path.'/registration/ticket/'.strval($id), 'GET');
+            $this->addHateoasLink('checkin', $path.'/registration/ticket/'.strval($id).'/checkin', 'PUT');
+            $this->addHateoasLink('lost', $path.'/registration/ticket/'.strval($id).'/lost', 'PUT');
+            $this->addHateoasLink('pickup', $path.'/registration/ticket/'.strval($id).'/pickup', 'PUT');
+            $this->addHateoasLink('email', $path.'/registration/ticket/'.strval($id).'/email', 'PUT');
+            $this->addHateoasLink('print', $path.'/registration/ticket/'.strval($id).'/print', 'PUT');
+        }
+
+    }
+
+
+    public function buildTicket($base, $ticket)
```

Comment:
This idiom is one of the few things that makes me wish we were using an ORM with model classes. Being able to put transformations on the model would be nice. Otherwise, I mostly don't miss it.

Reply:
Sorry missed these comments.  Yeah I have not used many ORM model classes so I am unfamiliar with the benefits and difficulties.